.github/workflows/deploy-book.yml

```diff
@@ -24,47 +23,55 @@ jobs:
         with:
           msg: Apply automatic formatting
 
-  deploy-book:
+  build-book:
     runs-on: ubuntu-latest
     defaults:
       run:
         shell: bash -l {0}  # required for conda env
     timeout-minutes: 45
     steps:
-    - uses: actions/checkout@v4
+      - uses: actions/checkout@v4
+
+      - name: Install graphviz
+        run: sudo apt install --yes graphviz graphviz-dev
+
+      - name: Setup python and install McStas
+        uses: mamba-org/setup-micromamba@v2
+        with:
+          micromamba-version: 2.0.4-0
+          environment-name: dmsc-school-env
+          create-args: >-
+            python=3.11.*
+            mcstas
 
-    - name: Install graphviz
-      run: sudo apt install --yes graphviz graphviz-dev
+      - name: Install pip dependencies
+        run: |
+          python -m pip install -r requirements.txt
 
-    - name: Setup python and install McStas
-      uses: mamba-org/setup-micromamba@v2
-      with:
-        micromamba-version: 2.0.4-0
-        environment-name: dmsc-school-env
-        create-args: >-
-          python=3.11.*
-          mcstas
+      - name: Build the book
+        run: |
+          jupyter-book build -W --keep-going .
 
-    - name: Install pip dependencies
-      run: |
-        python -m pip install -r requirements.txt
+      - name: Check links
+        run: |
+          jupyter-book build -W --keep-going --builder linkcheck .
```

Comment:
Can we disable the linkcheck if we are running from a push to main?
It might fail PRs but we might still want to forcibly push to main if some external website is down but we need to urgently update the notebooks?

Reply:
Is that the best solution? How about manually triggering the workflow and passing an input that skips linkcheck?